crates/block-producer/src/state_view/mod.rs

```diff
@@ -92,7 +95,12 @@ where
         // identify a set of unauthenticated input notes which are not in the store yet; we'll use
         // this set to verify that these notes are currently in flight (i.e., they are output notes
         // of one of the inflight transactions)
-        let tx_inputs = self.store.get_tx_inputs(candidate_tx).await?;
+        let mut tx_inputs = self.store.get_tx_inputs(candidate_tx).await?;
+        if let Some(inflight) = self.accounts_in_flight.read().await.get(&candidate_tx.account_id())
+        {
+            // Merge the inflight account state into the store state.
+            tx_inputs.account_hash = Some(*inflight);
+        }
```

Comment:
This feels a little hacky, but maybe just a better comment explaining why we are aggregating these is good enough?

Reply:
I think this should be OK as long as we document it here and also in `ensure_tx_inputs_constraints()` as there we no longer check just against the store but more generally against the latest known account state.

As a side note (and probably not for this PR): I find the name `TransactionInputs` really confusing as what we are getting are not really inputs into the transaction but various elements needed to verify transaction validity. I don't have a better suggestion though.

Reply:
Maybe something something `TransactionState`-like? Its sort of the initial (subset of all) state required by the transaction.